rpc/types_transaction.go

```diff
@@ -295,13 +295,42 @@ func (b BroadcastedInvokeV1Transaction) MarshalJSON() ([]byte, error) {
 	return json.Marshal(output)
 }
 
-type BroadcastedDeclareTransaction struct {
+type BroadcastedDeclareTransaction interface{}
+
+var _ BroadcastedDeclareTransaction = BroadcastedDeclareTransactionV1{}
+var _ BroadcastedDeclareTransaction = BroadcastedDeclareTransactionV2{}
+
+type BroadcastedDeclareTransactionV1 struct {
+	BroadcastedTxnCommonProperties
+	ContractClass DepcreatedContractClass `json:"contract_class"`
```

Comment:
There is a typo in the name of the type. I pointed it out in this [comment](https://github.com/NethermindEth/starknet.go/pull/293#discussion_r1311174035) from another PR, but it will need to be fixed here as well